exp/api/v1alpha3/azuremachinepool_types.go

```diff
@@ -51,6 +52,10 @@ type (
 
 	// AzureMachinePoolSpec defines the desired state of AzureMachinePool
 	AzureMachinePoolSpec struct {
+		// FailureDomains is the list of failure domains this MachinePool should be attached to, as defined in Cluster API.
+		// This relates to an Azure Availability Zone.
+		FailureDomains clusterv1.FailureDomains `json:"failureDomains,omitempty"`
```

Comment:
where is this consumed by vmss?

Reply:
If this were to follow the convention established by `MachineSpec.FailureDomain`, shouldn't this really be on `MachinePoolSpec`?

Reply:
I was preparing a different PR with changes in `vmss`. Should I include all of them in this PR?

Reply:
yes please, 2 commits but same PR would be ideal.

Reply:
oh also @fiunchinho, @devigned's comment above is on point. This should really go in the CAPI `MachinePool` spec. The reason it was in both `Machine` and `AzureMachine` was to allow for back compat with Availability Zones... see https://github.com/kubernetes-sigs/cluster-api-provider-azure/issues/679#issuecomment-638931845 

So actually I would expect the CAPZ change to _only_ be the `vmss` consumption.

Reply:
Instead of adding the field to `MachinePoolSpec` , shouldn't CAPZ use `MachinePoolSpec.Template.Spec.FailureDomain`?

Reply:
Oh I realized that the field is `FailureDomain *string`, and we need a list of `FailureDomains`. I'll try adding the field to `MachinePool` then.